config/src/config.cpp

```diff
@@ -198,3 +212,135 @@ config::Config::get<std::map<int, std::vector<int>>>(const std::string &key)
 
     return result;
 }
+
+template <>
+std::vector<std::vector<std::pair<std::string, int>>>
+config::Config::get<std::vector<std::vector<std::pair<std::string, int>>>>(const std::string &key)
+{
+    if (!json.isMember(key)) {
+        throw config::InvalidKeyException{key};
+    }
+
+    std::vector<std::vector<std::pair<std::string, int>>> result;
+    for (const auto &json_row : json[key]) {
+        if (!json_row.isArray()) {
+            throw config::InvalidValueException{
+                "get<std::vector<std::vector<std::pair<std::string, int>>>>"};
+        }
+
+        std::vector<std::pair<std::string, int>> row;
+        for (const auto &elem : json_row) {
+            row.push_back(convert_from_json<std::pair<std::string, int>>(elem));
+        }
+        result.push_back(row);
+    }
+
+    return result;
+}
+
+template <>
+std::vector<std::vector<int>>
+config::Config::get<std::vector<std::vector<int>>>(const std::string &key1, const std::string &key2)
```

Comment:
Det her API er ikke ret læsbar, det tog mig noget kravlen rundt i JSON filen at finde ud af hvad den laver. Jeg ville hellere kalde den noget a la `get_infomap_field` frem for bare `get` (og så skrive implementeringen efter det), da den er markant anderledes end de andre `get` specialiseringer. Hvis det føles ubehageligt så hiv det ud som en ekstern funktion med parametre `const std::string &key, const Config &config` hvor `key` indekserer ind i `robot_info_map` values.

Reply:
Vi er enige om, det ikke er så kønt, men det virker nu, og jeg gider ikke pille for meget i det.